py-rse/package-py.Rmd

```diff
@@ -482,450 +173,562 @@ without affecting the main installation (Figure \@ref(fig:py-rse-package-py-vir
-Successfully installed zipfpy-0.1
+```shell
+python setyp.py sdist
 ```
 
-Success!  If you take a look in `test_zipfpy/lib/python3.7/site-packages/`, you will
-now see the package with all the other site packages.
+```text
+running sdist
+running egg_info
+writing zipfs_law.egg-info/PKG-INFO
+writing dependency_links to zipfs_law.egg-info/dependency_links.txt
+writing entry points to zipfs_law.egg-info/entry_points.txt
+writing requirements to zipfs_law.egg-info/requires.txt
+writing top-level names to zipfs_law.egg-info/top_level.txt
+package init file 'zipfs_law/__init__.py' not found (or not a regular file)
+reading manifest file 'zipfs_law.egg-info/SOURCES.txt'
+writing manifest file 'zipfs_law.egg-info/SOURCES.txt'
+running check
+warning: check: missing required meta-data: url
 
-We can generate Zipf-distributed data to our hearts content:
+warning: check: missing meta-data: if 'author' supplied, 'author_email' must be supplied too
 
-```python
->>> import zipfpy
->>> import zipfpy.generate
->>> zipf_out = zipfpy.generate.make_zipf(5)
->>> print(zipf_out)
-[1.0, 0.5, 0.3333333333333333, 0.25, 0.2]
+creating zipfs_law-0.1
+creating zipfs_law-0.1/zipfs_law
+creating zipfs_law-0.1/zipfs_law.egg-info
+copying files to zipfs_law-0.1...
+copying README.md -> zipfs_law-0.1
+copying setup.py -> zipfs_law-0.1
+copying zipfs_law/collate.py -> zipfs_law-0.1/zipfs_law
+copying zipfs_law/countwords.py -> zipfs_law-0.1/zipfs_law
+copying zipfs_law/mymodule.py -> zipfs_law-0.1/zipfs_law
+copying zipfs_law/plotcounts.py -> zipfs_law-0.1/zipfs_law
+copying zipfs_law.egg-info/PKG-INFO -> zipfs_law-0.1/zipfs_law.egg-info
+copying zipfs_law.egg-info/SOURCES.txt -> zipfs_law-0.1/zipfs_law.egg-info
+copying zipfs_law.egg-info/dependency_links.txt -> zipfs_law-0.1/zipfs_law.egg-info
+copying zipfs_law.egg-info/entry_points.txt -> zipfs_law-0.1/zipfs_law.egg-info
+copying zipfs_law.egg-info/requires.txt -> zipfs_law-0.1/zipfs_law.egg-info
+copying zipfs_law.egg-info/top_level.txt -> zipfs_law-0.1/zipfs_law.egg-info
+Writing zipfs_law-0.1/setup.cfg
+creating dist
+Creating tar archive
+removing 'zipfs_law-0.1' (and everything under it)
 ```
 
-But if we continue and try to check the output, we'll find a problem:
+These distribution files can now be distributed via `pypi`,
+the standard Python repository for Python packages.
+There is a `testpypi` repository that lets us test putting packages on pypi 
+without having them appear on the standard repository;
+let's start with that.
+Go to [`https://test.pypi.org`](https://test.pypi.org),
+create an account,
+and confirm your email.
 
-```python
->>> import zipfpy.check
-Traceback (most recent call last):
-  File "<stdin>", line 1, in <module>
-  File "/Users/pterry/zipfpy/zipfpy/check.py", line 2, in <module>
-    from pytest import approx
-ModuleNotFoundError: No module named 'pytest'
-```
+The preferred tool for uploading packages to PyPI is called [`twine`](https://twine.readthedocs.io/en/latest/);
```

Comment:
PyPI is written without capital letters earlier on - maybe do with "PyPI" everywhere?

Also, do we put hyperlinks directly into the text like this or does there need to a be a new twine entry in `links.md`?